doc/reference/usermode/index.rst

```diff
@@ -41,15 +41,41 @@ For threads running in a non-privileged CPU state (hereafter referred to as
   memory that has an incompatible policy, such as attempting to write to a
   read-only area.
 
-  - Threads are automatically granted access to their own stack memory
-    region, and all other stacks are inaccessible.
+  - A user thread's access to memory is controlled by its memory domain
+    membership. A thread will always be a member of exactly one memory domain.
+    The memory domain represents regions of accessible physical memory
+    on MPU systems, and virtual address spaces on MMU systems.
 
-  - By default, program text and read-only data are accessible to all threads
-    on read-only basis, kernel-wide. This policy may be adjusted.
+  - Access to thread stack buffers will be controlled with a policy which
+    partially depends on the underlying memory protection hardware.
 
-  - If the optional "application memory" feature is enabled, then all
-    non-kernel globals defined in the application and libraries will be
-    accessible.
+    - A user thread will by default have read/write access to its own stack
+      buffer.
+
+    - A user thread will never by default have access to user thread stacks that
+      are not members of the same memory domain.
+
+    - A user thread will never by default have access to thread stacks owned
+      by a supervisor thread, or thread stacks used to handle system call
+      privilege elevations, interrupts, or CPU exceptions.
+
+    - A user thread may have read/write access to the stacks of other threads
+      in the same memory domain, depending on hardware.
+
+       - On MPU systems, threads may only access their own stack buffer.
+
+       - On MMU systems, threads may access any user thread stack in the same
+         memory domain.
+
+  - A user thread will by default have access to all of its necessary
+    program text. Whether this is all program text in the system, or a subset
+    thereof, will be configuration dependent.
+
+  - By default, read-only data is accessible to all threads on read-only basis,
+    kernel-wide. This policy may be adjusted.
```

Comment:
No 'private' read-only memory possible? This means if some confidential data is required, it always has to be RW?
Or, is this intended to cover the RO data in text segments, instead of a read-only memory domain partition?